clients/banking/src/components/MerchantProfilePaymentLinkArea.tsx

```diff
@@ -0,0 +1,292 @@
+import { AsyncData, Result } from "@swan-io/boxed";
+import { Link } from "@swan-io/chicane";
+import { useQuery } from "@swan-io/graphql-client";
+import { Box } from "@swan-io/lake/src/components/Box";
+import { Fill } from "@swan-io/lake/src/components/Fill";
+import { FocusTrapRef } from "@swan-io/lake/src/components/FocusTrap";
+import { FullViewportLayer } from "@swan-io/lake/src/components/FullViewportLayer";
+import { LakeButton } from "@swan-io/lake/src/components/LakeButton";
+import { LakeSearchField } from "@swan-io/lake/src/components/LakeSearchField";
+import { ListRightPanel } from "@swan-io/lake/src/components/ListRightPanel";
+import { PlainListViewPlaceholder } from "@swan-io/lake/src/components/PlainListView";
+import { Space } from "@swan-io/lake/src/components/Space";
+import { Toggle } from "@swan-io/lake/src/components/Toggle";
+import { LinkConfig } from "@swan-io/lake/src/components/VirtualizedList";
+import { spacings } from "@swan-io/lake/src/constants/design";
+import { isNotNullish, nullishOrEmptyToUndefined } from "@swan-io/lake/src/utils/nullish";
+import { useCallback, useMemo, useRef, useState } from "react";
+import { StyleSheet } from "react-native";
+import { P, match } from "ts-pattern";
+import {
+  MerchantPaymentLinkFiltersInput,
+  MerchantPaymentLinksDocument,
+  PaymentLinkFragment,
+} from "../graphql/partner";
+import { t } from "../utils/i18n";
+import { Router } from "../utils/routes";
+import { Connection } from "./Connection";
+import { ErrorView } from "./ErrorView";
+import { MerchantProfilePaymentLinkDetail } from "./MerchantProfilePaymentLinkDetail";
+import { MerchantProfilePaymentLinkNew } from "./MerchantProfilePaymentLinkNew";
+import { MerchantProfilePaymentLinksList } from "./MerchantProfilePaymentLinksList";
+
+const styles = StyleSheet.create({
+  containerMobile: { paddingTop: spacings[24], paddingHorizontal: spacings[24] },
+  filters: {
+    paddingHorizontal: spacings[24],
+    paddingTop: spacings[24],
+    paddingBottom: spacings[12],
+  },
+  filtersLarge: {
+    paddingHorizontal: spacings[40],
+  },
+  endFilters: {
+    flexGrow: 0,
+    flexShrink: 1,
+  },
+});
+
+const PER_PAGE = 20;
+
+type Props = {
+  accountMembershipId: string;
+  merchantProfileId: string;
+  params: { status: "Active" | "Archived"; search: string };
```

Comment:
By using [Chicane utility types](https://swan-io.github.io/chicane/utility-types), you can extract all types from your route directly, allowing you to type params with with `GetRouteParams<"AccountMerchantsProfilePaymentLinkArea">` instead.

This way, as the parent match `AccountMerchantsProfilePaymentLinkArea`, you can pass the whole params object.